### PR TITLE
Update/k8s client

### PR DESCRIPTION
**Pull request by punit-kulal**
Update k8s client to 0.22.12
- Changes in kubehandler interface to include context in accordance with k8 client interface changes.
- Update workqueue.Run to take context which would be passed to the handler functions.
- Bump module version to v2 since the default handler interface has been updated and is backward incompatible.
- Bumped golang version to 1.18

### CODE REVIEW

File: sample-controller/controller.go [hunk 1]
```diff
@@ -37,32 +38,33 @@ func main() {
...
-	loop := kubehandler.NewEventLoop("logger_queue")
+	loop := kubehandlerv2.NewEventLoop("logger_queue")
 
 	// Register all your handlers
 	loop.Register(loggingHandler)
 
 	// We're not handling signals for clean teardown. For production code, you
 	// probably want to do that
+	// We could also pass the context's done channel as well.
 	stopCh := make(chan struct{})
```
Comment (jitendra-gojek) [inline sample-controller/controller.go]: If we are not using this, this can be removed

File: workqueue_test.go [hunk 2]
```diff
@@ -26,46 +27,46 @@ func TestShouldEnqueueIntoTheUnderlyingWorkQueue(t *testing.T) {
 }
 
 func TestShouldCallRegisteredAddFuncWhenAddEventIsReceived(t *testing.T) {
-	workQueue := kubehandler.NewWorkQueue("WorkqueueTest2")
+	workQueue := kubehandlerv2.NewWorkQueue("WorkqueueTest2")
 	kind := "Foo"
 	addHandlerCalled := make(chan bool, 1)
 	stopChan := make(chan struct{}, 1)
```
Comment (jitendra-gojek) [inline workqueue_test.go]: remove this as well if not used

File: workqueue_test.go [hunk 3]
```diff
@@ -26,46 +27,46 @@ func TestShouldEnqueueIntoTheUnderlyingWorkQueue(t *testing.T) {
...
 	assert.True(t, <-addHandlerCalled)
 	close(stopChan)
 }
 
 func TestShouldCallRegisteredUpdateFuncWhenUpdateEventIsReceived(t *testing.T) {
-	workQueue := kubehandler.NewWorkQueue("WorkqueueTest3")
+	workQueue := kubehandlerv2.NewWorkQueue("WorkqueueTest3")
 	kind := "Foo"
 	updateHandlerCalled := make(chan bool, 1)
 	stopChan := make(chan struct{}, 1)
```
Comment (jitendra-gojek) [inline workqueue_test.go]: here as well

File: workqueue_test.go [hunk 4]
```diff
@@ -26,46 +27,46 @@ func TestShouldEnqueueIntoTheUnderlyingWorkQueue(t *testing.T) {
...
 	assert.True(t, <-updateHandlerCalled)
 	close(stopChan)
 }
 
 func TestShouldCallRegisteredDeleteFuncWhenDeleteEventIsReceived(t *testing.T) {
-	workQueue := kubehandler.NewWorkQueue("WorkqueueTest4")
+	workQueue := kubehandlerv2.NewWorkQueue("WorkqueueTest4")
 	kind := "Foo"
 	deleteHandlerCalled := make(chan bool, 1)
 	stopChan := make(chan struct{}, 1)
```
Comment (jitendra-gojek) [inline workqueue_test.go]: here as well

**Review by jitendra-gojek — CHANGES_REQUESTED** [review]
Assign it to karthik also for review

**Comment by buildAI** [comment]
The PR looks fine to me.

File: sample-controller/controller.go (hunk 1 above)
Comment (punit-kulal) [reply sample-controller/controller.go]: Have removed the usage of stopCh and started using context

File: workqueue_test.go (hunk 2 above)
Comment (punit-kulal) [reply workqueue_test.go]: Have removed the usage of stopCh and started using context

File: workqueue_test.go (hunk 4 above)
Comment (punit-kulal) [reply workqueue_test.go]: Have removed the usage of stopCh and started using context

**Review by ruhcom — APPROVED** [review]
LGTM

**Review by jitendra-gojek — APPROVED** [review]
LGTM..

**Pull request merged**